examples/typescript-react/js/app.tsx

```diff
@@ -0,0 +1,181 @@
+/*jshint quotmark:false */
+/*jshint white:false */
+/*jshint trailing:false */
+/*jshint newcap:false */
+/*global React, Router*/
+
+/// <reference path="../typings/tsd.d.ts" />
+/// <reference path="./interfaces.d.ts"/>
+
+declare var Router;
+
+import { TodoModel } from "./TodoModel";
+import { TodoFooter } from "./footer";
+import { TodoItem } from "./todoItem";
+import { ALL_TODOS, ACTIVE_TODOS, COMPLETED_TODOS, ENTER_KEY } from "./constants";
+
+class TodoApp extends React.Component<IAppProps, IAppState> {
+
+  public state : IAppState;
+
+  constructor(props : IAppProps) {
+    super(props);
+    this.state = {
+      nowShowing: ALL_TODOS,
+      editing: null
+    };
+  }
+
+  public componentDidMount() {
+    var setState = this.setState;
+    var router = Router({
+      '/': setState.bind(this, {nowShowing: ALL_TODOS}),
+      '/active': setState.bind(this, {nowShowing: ACTIVE_TODOS}),
+      '/completed': setState.bind(this, {nowShowing: COMPLETED_TODOS})
+    });
+    router.init('/');
+  }
+
+  public handleNewTodoKeyDown(event : __React.KeyboardEvent) {
```

Comment:
Nit, but I believe `event: __React.KeyboardEvent` would be more consistent with the other typescript apps.